enhancements/etcd/scaling-etcd-with-raft-learners.md

```diff
@@ -0,0 +1,343 @@
+---
+title: Scaling etcd with Raft Learners
+authors:
+- "@hexfusion"
+  reviewers:
+- "@deads2k"
+- "@lilic"
+- "@hasbro17"
+- "@joelspeed"
+- "@jeremyeder"
+  approvers:
+- "@mfojtik"
+  creation-date: 2021-10-04
+  last-updated: 2021-10-04
+  status: implementable
+  see-also: []
+  replaces: []
+  superseded-by: []
+---
+
+# Scaling etcd with Raft Learners
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+  [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Over time as clusters live longer and workloads grow the ability to scale the control-plane and replace failed nodes
+becomes a critical part of the admins maintenance overhead. Today the `cluster-etcd-operator` manages scaling up of
+the etcd cluster. To provide the foundation for initiatives such as scale down and vertical control plane scaling[1].
+The `cluster-etcd-operator` must ensure proper safety mechanisms exist to adjust membership of the etcd cluster.
+
+Introduced in etcd v3.4 the raft learner[2] provides mitigations which reduces quorum and
+stability issues during scaling. A learner is essentially an etcd member which is non voting thus can not impact 
+quorum but like other members will receive log replications from the leader.
+
+This enhancement proposes:
+- Replacing the default scale up performed by the cluster-etcd-operator to use raft learners.
+- Deprecation and removal of the current `discover-etcd-initial-cluster`[3] command and replacing its
+functionality with the existing `etcdenvvar` controller and subcommands of the `cluster-etcd-operator`. By
+eliminating this code it also removes the teams largest carry against upstream etcd.
+- Ensuring that scale up only happens on control-plane nodes which have a quorum-guard pod monitoring health.
+- Adding a flag to etcd which allows configuration of maximum learners in cluster (--max-learners). Today the max is 1.
+
+POC: Functional proof of concept: https://github.com/openshift/cluster-etcd-operator/pull/682
+
+[1] https://issues.redhat.com/browse/OCPPLAN-5712
+
+[2] https://etcd.io/docs/v3.4/learning/design-learner/
+
+[3] https://github.com/openshift/etcd/blob/openshift-4.10/openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go
+
+## Motivation
+
+Great demand exists to provide flexible automated scaling of the control-plane in the same way the worker
+nodes do today.
+
+### Goals
+
+- provide safe scale up of etcd cluster using raft learners.
```

Comment:
Vertical scaling is a capability we need for managed platforms.  Is this scale up in support of changing instance types for the underlying master?  And can be followed by a graceful and clean scale down of older masters?